packages/lodestar/src/util/queue.ts

```diff
@@ -25,43 +25,52 @@ type JobQueueItem<T extends Job = Job> = {
   reject: (reason?: unknown) => void;
 };
 
-export class JobQueue {
+type JobQueueOpts = {
   queueSize: number;
-  currentSize: number;
   signal: AbortSignal;
-  queue: Pushable<JobQueueItem>;
-  processor: Promise<void>;
-  constructor({queueSize, signal}: {queueSize: number; signal: AbortSignal}) {
-    this.queueSize = queueSize;
-    this.currentSize = 0;
-    this.signal = signal;
-    this.queue = pushable();
-    this.processor = pipe(this.queue, async (source) => {
+  /**
+   * Called when a job resolves or rejects.
+   * Returns the total miliseconds ellapsed from job start to done
+   */
+  onJobDone?: (data: {ms: number}) => void;
+};
+
+export class JobQueue {
+  private currentSize = 0;
+  private queue = pushable<JobQueueItem>();
+  private opts: JobQueueOpts;
+
+  constructor(opts: JobQueueOpts) {
+    this.opts = opts;
+    void pipe(this.queue, async (source) => {
       for await (const job of source) {
         await this.processJob(job);
       }
     });
   }
 
   async processJob({job, resolve, reject}: JobQueueItem): Promise<void> {
-    if (this.signal.aborted) {
+    if (this.opts.signal.aborted) {
       reject(new QueueError({code: QueueErrorCode.QUEUE_ABORTED}));
     } else {
+      const start = Date.now();
       try {
         const result = await job();
         resolve(result);
       } catch (e) {
         reject(e);
+      } finally {
+        this.currentSize--;
```

Comment:
previously, `this.currentSize` was decremented in both the if and else.

Reply:
Fixed